src/main/java/com/rackspace/ceres/app/services/MetadataService.java

```diff
@@ -127,75 +107,57 @@ public MetadataService(ReactiveCqlTemplate cqlTemplate,
 
   public Publisher<?> storeMetadata(String tenant, String seriesSetHash, Metric metric) {
     final CompletableFuture<Boolean> result = seriesSetExistenceCache.get(
-            new SeriesSetCacheKey(tenant, seriesSetHash),
-            (key, executor) ->
-                    cassandraTemplate.exists(
-                                    query(
-                                            where(COL_TENANT).is(tenant),
-                                            where(COL_SERIES_SET_HASH).is(seriesSetHash)
-                                    ),
-                                    SeriesSetHash.class
-                            )
-                            .name("metadataExists")
-                            .metrics()
-                            .retryWhen(appProperties.getRetryInsertMetadata().build())
-                            .doOnError(t -> log.error("Exception in exists check "+t.getMessage()))
-                            .doOnNext(exists -> {
-                              if (exists) {
-                                cassandraHit.increment();
-                              } else {
-                                cassandraMiss.increment();
-                              }
-                            })
-                            // not cached in cassandra?
-                            .flatMap(exists -> exists ?
-                                    Mono.just(true) :
-                                    // not cached, so store the metadata to be sure
-                                    storeMetadataInCassandra(tenant, seriesSetHash, metric)
-                                        .and(elasticSearchService.saveMetricToES(tenant, metric))
-                                            .map(it -> true))
-                            .toFuture()
+        new SeriesSetCacheKey(tenant, seriesSetHash),
```

Comment:
Can't we just delete the cache and the storing in cassandra?

Reply:
Storing into ES will only be called only if it's not in cache otherwise we will be making extra index api calls for each and every metrics coming to ceres app.